related-files.el

```diff
@@ -234,35 +247,71 @@ ones.  Look at `related-files-jump' and `related-files-make' if you don't
 want to mix existing and non-existing files in the same list..
 
 If JUMPERS is not provided, use `related-files-jumpers'.  If
-CURRENT-PLACE is not provided, use the function
-`buffer-file-name'.
+CURRENT-PLACES is not provided, use
+`related-files-current-place-finders'.
 
 Interactively, a numeric prefix argument selects the jumper at
 the specified position (zero-based index) in `related-files-jumpers'."
   (interactive (list (when (numberp current-prefix-arg)
                        (list (seq-elt related-files-jumpers current-prefix-arg)))))
-  (related-files--jump-or-make jumpers current-place
+  (related-files--jump-or-make jumpers current-places
                          :include-existing-places t
                          :include-non-existing-places t))
 
 
-;;; Jumpers Public API
+;;; Jumpers and Places Public API
 
 (cl-defgeneric related-files-apply (jumper place)
   "Apply JUMPER to PLACE and return related places or nil.
 
 PLACE is a filename and the result must be a possibly-empty list
-of filenames.
+of filenames.")
 
-The default implementation allows JUMPER to be a function.  The
-function can return either a single place or a possibly-empty
-list of places."
+(cl-defmethod related-files-apply ((jumper symbol) place)
+"Call JUMPER on PLACE."
   (funcall jumper place))
 
 (cl-defgeneric related-files-get-filler (jumper)
   "Return a filler associated with JUMPER."
   (get jumper 'related-files-filler))
 
+(cl-defgeneric related-files-place-exists-p (place)
+  "Return non-nil if PLACE exists.
+
+The default implementation assumes PLACE is a filename."
+  (file-exists-p place))
+
+(cl-defmethod related-files-place-exists-p ((place buffer))
```

Comment:
why is there no implementation of this method for filenames?

Reply:
When I wrote this, I was still following the earlier practice of writing a defgeneric with non-specialised arguments, and with a default iimplementation which assumed the place was a filename (so the implementation for filenames is just the default implementation).

Do you think it would be better to completely separate the definitions for all these generics/methods (`place-exists-p` and`goto-place`

Reply:
I've no strong feeling about this so pick whatever you want. It feels to me that the default for `related-files-place-exists-p` should probably return `nil` and the default for `related-files-goto-place` should throw.